sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -718,39 +630,4 @@ trait ScalaReflection {
         throw new UnsupportedOperationException(s"Schema for type $other is not supported")
     }
   }
-
-  def typeOfObject: PartialFunction[Any, DataType] = {
```

Comment:
This method is only used in tests, so I think we should remove it and its related tests.